resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicDriverFeatureStep.scala

```diff
@@ -73,7 +69,7 @@ private[spark] class BasicDriverFeatureStep(conf: KubernetesDriverConf)
   private val memoryOverheadFactor = if (conf.contains(DRIVER_MEMORY_OVERHEAD_FACTOR)) {
     conf.get(DRIVER_MEMORY_OVERHEAD_FACTOR)
   } else {
-    defaultOverheadFactor
+    defaultOverheadFactor.getOrElse(conf.get(DRIVER_MEMORY_OVERHEAD_FACTOR))
```

Comment:
Let me know if this logic makes sense for get the new config if defined, otherwise deprecated config if defined, otherwise default value of new config

Reply:
The new logic looks reasonable to me, @Kimahriman .